app/javascript/packages/memorable-date/index.ts

```diff
@@ -0,0 +1,386 @@
+      return maxErrorMessage;
+    }
+    // Set another range error if applicable and messaging is available
+    return rangeErrors.find(
+      ({ min: rangeMin, max: rangeMax }) =>
+        (rangeMin && date < new Date(rangeMin)) || (rangeMax && date > new Date(rangeMax)),
+    )?.message;
+  }
+
+  /**
+   * @param attrName Name of attribute on this element containing a parseable date
+   * @returns Date Javascript object parsed from the given attribute or null if it's misssing/invalid
+   */
+  private getDateAttribute(attrName: string): Date | null {
+    const raw = this.getAttribute(attrName);
+    if (raw === null) {
+      return null;
+    }
+
+    const value = Date.parse(raw);
+    if (Number.isNaN(value)) {
+      return null;
+    }
+    return new Date(value);
+  }
+
+  /**
+   * Set a custom error message on the given fields for use by a containing
+   * ValidatedFieldElement.
+   *
+   * @param message Error message to display
+   * @param fields Fields to which the error message applies
+   */
+  private setValidity(message: string, ...fields: HTMLInputElement[]): void {
+    const { allInputs } = this;
+    allInputs.forEach((field) => {
+      if (fields.includes(field)) {
+        field.setCustomValidity(message);
+      } else {
+        field.setCustomValidity('');
+      }
+    });
+  }
+
+  /**
+   * @param errs Hash for looking up error messages
+   * @returns Function to show errors for missing values on a group of elements
+   */
+  private checkMissingValues(errs: MemorableDateErrorMessageLookup) {
+    return (fields: { [k: string]: HTMLInputElement | undefined }): boolean => {
+      const message = errs[`missing_${Object.keys(fields).join('_')}`];
+      const fieldValues = Object.values(fields);
+      if (fieldValues.every((field) => !field?.value)) {
+        message && this.setValidity(message, ...(fieldValues as HTMLInputElement[]));
+        return true;
+      }
+      return false;
+    };
+  }
+
+  /**
+   * @param errs Hash for looking up error messages
+   * @returns Function to show errors for invalid values on a group of elements
+   */
+  private checkFieldsInvalid(errs: MemorableDateErrorMessageLookup) {
+    return (fields: { [k: string]: HTMLInputElement | undefined }): boolean => {
+      const message = errs[`invalid_${Object.keys(fields).join('_')}`];
+      const fieldValues = Object.values(fields);
+      if (fieldValues.every((field) => field?.validity.patternMismatch)) {
+        message && this.setValidity(message, ...(fieldValues as HTMLInputElement[]));
+        return true;
+      }
+      return false;
+    };
+  }
+
+  /**
+   * @param entry Entry to check from Object.entries call
+   * @returns True if the given entry is a valid mapping for memorable date errors
+   */
+  private isValidErrorMessage(entry: [string, any]): entry is [MemorableDateErrorMessage, string] {
+    return (
+      Array.isArray(entry) &&
+      MemorableDateErrorMessage[entry[0]] === entry[0] &&
```

Comment:
Following-up on my previous comment about converting the enum to a type, I see now that we are referencing those values.

I've gotta wonder though, do we really need this validation? What's the scenario where we would be extracting invalid error messages, and what would be the consequences if we allowed that?

Reply:
At this point it's more conveying information about what the lookup stores than changing behavior. I can loosen this check to only verify that the error message is a `string`.

Reply:
Updated type guard to only check that the value is a `string`.

Reply:
> At this point it's more conveying information about what the lookup stores than changing behavior. I can loosen this check to only verify that the error message is a `string`.

Isn't that what our type interfaces are conveying?

https://github.com/18F/identity-idp/blob/322a0b97d2e1edd8ca703341081564482ff0e237/app/javascript/packages/memorable-date/index.ts#L34-L41

I don't think it's a blocker, I just think some of the validation here isn't necessary, and by removing the extra code it might be simpler to follow the logic of the component.